ampe.c

```diff
@@ -112,6 +112,9 @@ static inline void fsm_restart(struct candidate *cand)
     set_link_state(cand, PLINK_LISTEN);
     cand->my_lid = cand->peer_lid = cand->reason = 0;
     cand->retries = 0;
+    memset(cand->mtk, 0, sizeof(cand->mtk));
+    memset(cand->igtk, 0, sizeof(cand->igtk));
```

Comment:
Can you add?
`cand->has_igtk = false;`

Reply:
OK I made this change, but I also moved these things to peer_ampe_init() which looked like a better place (it still gets called after fsm_restart() due to my_lid = 0 assignment here).